### PR TITLE
🏃 Add ci-apidiff job

### PR DESCRIPTION
**Pull request by vincepri**
Signed-off-by: Vince Prignano <vincepri@vmware.com>

### CODE REVIEW

File: scripts/ci-apidiff.sh [hunk 1]
```diff
@@ -0,0 +1,28 @@
+#!/bin/bash
+
+# Copyright 2018 The Kubernetes Authors.
```
Comment (ncdc) [inline scripts/ci-apidiff.sh:3]: Is this accurate or should it be 2020?

Comment (vincepri) [reply scripts/ci-apidiff.sh]: copy pasted it, fixed

File: scripts/ci-apidiff.sh [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+set -o pipefail
+
+REPO_ROOT=$(dirname "${BASH_SOURCE[0]}")/..
+
+APIDIFF="hack/tools/bin/go-apidiff"
+
+cd "${REPO_ROOT}" && make ${APIDIFF}
+echo "*** Running go-apidiff ***"
+
+${APIDIFF} "${PULL_BASE_SHA}"
```
Comment (detiber) [inline scripts/ci-apidiff.sh:28]: Any particular reason for a separate script here over having a make target for running go-apidiff?

Comment (vincepri) [reply scripts/ci-apidiff.sh:28]: This wouldn't work in local environments, so I opted to keep it in the script file and just have the Makefile call the binary directly

Comment (detiber) [reply scripts/ci-apidiff.sh:28]: makes sense, thanks for the clarification

**Pull request merged**